lib/hotbar.js

```diff
@@ -19,40 +44,112 @@ class HotBar extends LitElement {
       this.reloadHotbarSelected(newSlot)
     })
 
-    this.bot.inventory.on('updateSlot', (slot, oldItem, newItem) => {
+    this.bot.inventory.on('updateSlot', async (slot, oldItem, newItem) => {
       if (slot >= this.bot.inventory.hotbarStart + 9) return
       if (slot < this.bot.inventory.hotbarStart) return
 
       const sprite = newItem ? invsprite[newItem.name] : invsprite.air
-      const slotImage = this.shadowRoot.getElementById('hotbar-' + (slot - this.bot.inventory.hotbarStart))
-      slotImage.style['background-position-x'] = `-${sprite.x * 2}px`
-      slotImage.style['background-position-y'] = `-${sprite.y * 2}px`
-      slotImage.innerHTML = newItem?.count > 1 ? newItem.count : ''
+      const slotImage = this.shadowRoot.getElementById('hotbar-' + (slot - this.bot.inventory.hotbarStart)).firstElementChild
+
+      const icon = await getItemIcon(sprite)
+      slotImage.innerHTML = ''
+      slotImage.appendChild(await icon)
+
+      slotImage.nextElementSibling.innerHTML = newItem?.count > 1 ? newItem.count : ''
     })
   }
 
   async reloadHotbar () {
     for (let i = 0; i < 9; i++) {
       const item = this.bot.inventory.slots[this.bot.inventory.hotbarStart + i]
       const sprite = item ? invsprite[item.name] : invsprite.air
-      const slotImage = this.shadowRoot.getElementById('hotbar-' + i)
-      slotImage.style['background-position-x'] = `-${sprite.x * 2}px`
-      slotImage.style['background-position-y'] = `-${sprite.y * 2}px`
-      slotImage.innerHTML = item?.count > 1 ? item.count : ''
+      const slotImage = this.shadowRoot.getElementById('hotbar-' + i).firstElementChild
+
+      const icon = await getItemIcon(sprite)
+      slotImage.innerHTML = ''
+      slotImage.appendChild(await icon)
+
+      slotImage.nextElementSibling.innerHTML = item?.count > 1 ? item.count : ''
     }
   }
 
   async reloadHotbarSelected (slot) {
     const item = this.bot.inventory.slots[this.bot.inventory.hotbarStart + slot]
-    const planned = (20 * 4 * slot) + 'px'
-    this.shadowRoot.getElementById('hotbar-highlight').style.marginLeft = planned
+
+    const hotbarSlots = this.shadowRoot.querySelectorAll('.hotbar-slot')
+    for (let i = 0; i < hotbarSlots.length; i++) {
+      const hotbarSlot = hotbarSlots[i]
+
+      hotbarSlot.classList.remove('selected')
+      if (slot === i) hotbarSlot.classList.add('selected')
+    }
+
     this.bot.setQuickBarSlot(slot)
     this.activeItemName = item?.displayName
     const name = this.shadowRoot.getElementById('hotbar-item-name')
     name.classList.remove('hotbar-item-name-fader')
     setTimeout(() => name.classList.add('hotbar-item-name-fader'), 10)
   }
 
+  async reloadHealth (health) {
```

Comment:
Could you avoid adding more manual dom manipulation and instead use lit html templates in render function ?

Reply:
wdym

Reply:
have a seperate lit element for the health, food and experience bar?

Reply:
I mean you don't do any this.shadowRoot.getElementById
And instead put everything in render function